earthfile2llb/converter.go

```diff
@@ -297,6 +297,16 @@ func (c *Converter) CopyClassical(ctx context.Context, srcs []string, dest strin
 			dest))
 }
 
+func (c *Converter) RunLocal(ctx context.Context, args []string) {
+	finalArgs := withShellAndEnvVars(args, []string{}, true, false)
+	c.mts.Final.MainState = c.mts.Final.MainState.Run(
+		llb.Args(finalArgs),
+		llb.AddMount("/run_on_localhost_hack", llb.Scratch()),
+		llb.IgnoreCache,
+		llb.WithCustomNamef("%sRUN %s", c.vertexPrefix(), strings.Join(args, " ")),
```

Comment:
I wonder if we should mark the output in a special way when it's executing on the local machine.

Reply:
that's a good idea, maybe `LOCAL RUN` or `RUN LOCAL`?

e.g. 
```
alex@mah:~/gh/earthly/earthly$ ./build/linux/amd64/earthly --push +test-local 
           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
golang:1.13-alpine3.11 | --> Load metadata linux/amd64
      busybox:1.32.1 | --> Load metadata linux/amd64
         +test-local | --> FROM busybox:1.32.1
         +test-local | [██████████] resolve docker.io/library/busybox:1.32.1@sha256:c5439d7db88ab5423999530349d327b04279ad3161d7596d2126dfb5b02bfd1f ... 100%
         +test-local | --> LOCAL RUN cat /proc/1/cgroup | grep '^[0-9]\+:cpuset:/$'
         +test-local | 12:cpuset:/
              output | --> exporting outputs
              output | [██████████] sending tarballs ... 100%
=========================== SUCCESS ===========================
         +test-local | 12:cpuset:/
         +test-local | --> LOCAL RUN --push touch /tmp/earthly-test-local
```

Reply:
Or even `+test-local *local* | --> RUN cat ...` (this is similar to how we print *failed*). We can then even use a special color (orange?) for local commands to draw more attention.

Reply:
that definitely makes it stand out more.
```
alex@mah:~/junk/local$ /home/alex/gh/earthly/earthly/build/linux/amd64/earthly +both 
           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)
       alpine:latest | --> Load metadata linux/amd64
      busybox:1.32.1 | --> Load metadata linux/amd64
              +step1 | --> FROM busybox:1.32.1
              +step2 | --> FROM alpine:latest
              +step1 | [██████████] resolve docker.io/library/alpine:latest@sha256:d9a7354e3845ea8466bb00b22224d9116b183e594527fb5b6c3d30bc01a20378 ... 100%
              +step1 | [██████████] resolve docker.io/library/busybox:1.32.1@sha256:c5439d7db88ab5423999530349d327b04279ad3161d7596d2126dfb5b02bfd1f ... 100%
      +step1 *local* | --> RUN whoami > myuser
      +step1 *local* | --> RUN pwd
      +step1 *local* | /home/alex/junk/local
      +step1 *local* | --> RUN cat /proc/1/cgroup
      +step1 *local* | 12:cpuset:/
      +step1 *local* | 11:perf_event:/
      +step1 *local* | 10:devices:/
      +step1 *local* | 9:blkio:/
      +step1 *local* | 8:freezer:/
      +step1 *local* | 7:cpu,cpuacct:/
      +step1 *local* | 6:net_cls,net_prio:/
      +step1 *local* | 5:rdma:/
      +step1 *local* | 4:memory:/
      +step1 *local* | 3:pids:/
      +step1 *local* | 2:hugetlb:/
      +step1 *local* | 1:name=systemd:/init.scope
      +step1 *local* | 0::/init.scope
              +step2 | *cached* --> RUN cat /proc/1/cgroup
              +step2 | *cached* --> RUN sleep 10
              +step2 | *cached* --> RUN whoami > myuser
              +step2 | *cached* --> RUN cat myuser
              output | --> exporting outputs
              output | [██████████] sending tarballs ... 100%
=========================== SUCCESS ===========================
              +step1 | Did not execute push command RUN --push echo push flag1. Use earthly --push to enable pushing
              +step2 | Did not execute push command RUN --push echo push flag2. Use earthly --push to enable pushing
```

maybe we could leave out the orange feature for another PR (it'll also be a good first issue for anyone wanting to contribute).